java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/ClientCredentialsGrantTokenRequestBuilder.java

```diff
@@ -78,12 +91,19 @@ enum ConfigProperty implements ConfigService.ConfigProperty {
 
     @Override
     public Set<ConfigService.ConfigProperty> getRequiredConfigProperties() {
-        return Set.of(
-            OAuthRefreshTokenSourceAuthStrategy.ConfigProperty.REFRESH_ENDPOINT, //should be redundant with strategy, but nonetheless req'd here too
-            OAuthRefreshTokenSourceAuthStrategy.ConfigProperty.CLIENT_ID,
-            ConfigProperty.PRIVATE_KEY_ID,
-            ConfigProperty.PRIVATE_KEY
-        );
+        if (getCredentialsType() == CredentialFlowType.CLIENT_SECRET) {
```

Comment:
Merged both client credentials in old class to ensure compatibility with old versions @eschultink